registrazione.aspx.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+public partial class registrazione : System.Web.UI.Page
+{
+    protected void Page_Load(object sender, EventArgs e)
+    {
+
+    }
+
+    protected void btnInvia_Click(object sender, EventArgs e)
+    {
+        // controlli formali
+        if (string.IsNullOrEmpty(txtEmail.Text))
+        {
+            ScriptManager.RegisterClientScriptBlock(this, GetType(), "ATTENZIONE", "alert('Dati non validi')", true);
+            return;
+        }
+
+        // mi salvo la textbox come variabili
+        string mail = txtEmail.Text.Trim();
+
+        UTENTI u = new UTENTI();
+        u.email = mail;
+
+        // controllo se l'utente non sia già registrato
+        if (u.Registrato() == true)
+        {
+            ScriptManager.RegisterClientScriptBlock(this, GetType(), "ATTENZIONE", "alert('Utente già registrato')", true);
+            return;
+        }
+
+        // mando la mail di conferma all'indirizzo dell'utente con il codice di conferma generato casualmente
+        Random rnd = new Random();
+        string rndCodice = rnd.Next(100000, 999999).ToString();
+
```

Comment:
Sarebbe stato meglio prima inserire l'utente nel DB e poi mandare la mail, magari racchiudendo la seconda operazione in un try catch per mostrare a schermo la pass temporanea se l'invio mail non riesce